### DISCUSSION TITLE
rust 使用 ed25519 生成秘钥

### DISCUSSION THREAD

**Discussion by v1xingyue** [CreatorsDAO/rust-co-learn#9] [Q&A] [answered]
我想使用 rust 生成一堆  ed25519 的秘钥对，rust 里 使用的是  [ed25519-dalek](https://github.com/dalek-cryptography/ed25519-dalek)。
实例代码如下: 

```rust
extern crate rand;
extern crate ed25519_dalek;

use rand::rngs::OsRng;
use ed25519_dalek::Keypair;
use ed25519_dalek::Signature;

let mut csprng = OsRng{};
let keypair: Keypair = Keypair::generate(&mut csprng);
```

我在我电脑上编译的时候，报错如下: 

![image](https://github.com/CreatorsDAO/rust-co-learn/assets/974169/7a5114db-93e0-4489-8a68-7192a1c3eb51)

似乎是 OsRng 的问题。
rust 版本: rustc 1.70.0 (90c541806 2023-05-31) , 系统环境:osx M1 

版本信息: 

ed25519-dalek = "1"
rand = "0.8.5"

尝试更换过其他的rand,依然报错，怎么找一个符合条件的 Rng ?

**Comment by DaviRain-Su** [comment, marked as answer]
https://github.com/DaviRain-Su/all-in-one-rust/blob/main/ed25519-example/src/lib.rs

**Comment by zlemon819** [comment]
rand版本换一下就行了   ed25519-dalek v1.0.1 依赖rand0.7.0

[dependencies]
ed25519-dalek = "1"
rand = "0.7.0"
